### PR TITLE
`Programming exercises`: Remove the legacy README.md handling

### LINKED ISSUE #9214: `README.md` not shown in online code editor
### Describe the bug

If a student repository includes a `README.md` file, it is filtered out in the file tree of the online code editor. 
This appears to be some compatibility layer to a legacy type of programming exercises. (See [here](https://github.com/ls1intum/Artemis/blob/1a6169f070398ce4145e232f2a8e935ab4704e06/src/main/webapp/app/exercises/programming/shared/code-editor/file-browser/code-editor-file-browser.component.ts#L518)). 
New exercises that actually intend for some readme are hindered this filtering. 

### To Reproduce

1. Create an exercise.
2. Start exercise and commit a file called `README.md`.
3. Open either the online code editor or viewer.
4. The file will not be visible

### Expected behavior

The file is visible.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

7.5.0

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by pzdr7**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

Very long ago (#381), the problem statement of programming exercises was stored in the `README.md` file of the assignment repository. This meant that the online code editor had to filter out this file to prevent modifications. We no longer use this mechanism, so we can remove this restriction.

### Description
Closes #9214.

- Treats README.md like any other file in the online code editor
- No longer uses README.md as a backup for the problem statement of programming exercises
- Improved the function coverage of the file browser component because it was failing 

### Steps for Testing
Prerequisites:
- 1 Programming exercise with a `README.md` at the root level of the template repository. Enable the online code editor for this exercise.
- 1 Student

1. As the student, navigate to the exercise.
2. Open the online code editor.
3. Verify that the `README.md` file is shown and that you can edit it.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Only trivial changes. Coverage of the CodeEditorFileBrowserComponent improves to 95% (roughly +9%p)

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| code-editor-file-browser.component.ts | 95.04% | ✅ |
### Screenshots

No UI changes.

## Summary by CodeRabbit

- **New Features**
	- README.md files are now visible in the file browser, enhancing the user experience.
	- Streamlined instruction retrieval process focuses solely on problem statements, improving efficiency.

- **Bug Fixes**
	- Fixed logic to handle file and folder management, including creation, renaming, and deletion confirmation modals.

- **Documentation**
	- Improved comments and clarity regarding the purpose and behavior of various components. 

- **Tests**
	- Enhanced test cases for the file browser and programming exercise instruction components, focusing on new functionalities and dependencies.

### CODE REVIEW

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts [hunk 1]
```diff
@@ -138,84 +130,11 @@ describe('ProgrammingExerciseInstructionComponent', () => {
         expect(subscribeForLatestResultOfParticipationStub).toHaveBeenCalledOnce();
         expect(subscribeForLatestResultOfParticipationStub).toHaveBeenCalledWith(participation.id, true, exercise.id);
         expect(comp.participationSubscription).not.toEqual(oldSubscription);
+        flush();
         expect(comp.isInitial).toBeTrue();
-    });
-
-    it('should try to fetch README.md from assignment repository if no problemStatement was provided', () => {
```
Comment (pzdr7) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts:144]: As of this PR, the tests removed here are obsolete

**Review by SimonEntholzer — APPROVED** [review]
Tested on Ts3, README.md shows up for the student

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me

**Review by BaumiCoder — COMMENTED** [review]
Code looks good to me.

Only one little question to this topic: Old exercises have their problem statement in a README.md. Then this approach was replaced, but the README.md was still a fallback option for the problem statement. When was a migration of the README.md files into the new approach (the database)? Without such a migration this PR could break old exercises, isn't it?

**Comment by pzdr7** [comment]
@BaumiCoder I asked Stephan about this — the migration was several years ago.

We already have a lot of features in Artemis that rely on the problemStatement stored in the DB (grading, tasks, etc.) that wouldn't work with the old README.md approach, so this change wouldn't break any exercises that aren't already broken

**Review by BBesrour — APPROVED** [review]
tested on ts1, lgtm

**Review by BaumiCoder — APPROVED** [review]
Code

**Pull request merged**